### PR TITLE
OSSM-1343: clang build fix for s390x

### PR DESCRIPTION
**Pull request by knm3000**
Fixes https://issues.redhat.com/browse/OSSM-1065

### CODE REVIEW

File: envoy.bazelrc [hunk 1]
```diff
@@ -43,7 +43,7 @@ build:sanitizer --test_tag_filters=-no_san
 
 # Common flags for Clang
 build:clang --action_env=BAZEL_COMPILER=clang
-build:clang --linkopt=-fuse-ld=lld
+build:clang --linkopt=-fuse-ld=gold
```
Comment (oschaaf) [inline envoy.bazelrc]: This change doesn't look specific to s390.. is that on purpose?

Comment (jwendell) [reply envoy.bazelrc]: yes, see the slack conversation

Comment (oschaaf) [reply envoy.bazelrc]: I see, got it, resolving this.

**Review by jwendell — APPROVED** [review]

**Pull request merged**